server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2702,6 +2702,9 @@ private void ensureSoftDeletesEnabled() {
     @Override
     public Translog.Snapshot newChangesSnapshot(String source, MapperService mapperService,
                                                 long fromSeqNo, long toSeqNo, boolean requiredFullRange) throws IOException {
+        if (source.equals(HistorySource.TRANSLOG.name())) {
```

Comment:
Why do you need this change in source? Why not extend the engine in your plugin?

Reply:
1. We will need to dynamically load the engine every time and 
2. These changes are dependent on the core Tlog pruning changes.

Reply:
1. Not sure why you would need dynamic engine loading. It can be done without as well. 
2. Tlog pruning is a mechanism to avoid pruning tlog based on retention lease.

Reply:
Any reason to move only fetch to the plugin as all the changes are still applicable under core?

Reply:
I want to minimize the changes in core due to this change as this is a stop gap solution till we are confident that this is what we want to use for peer recoveries as well or if we want to make Lucene optimizations and use peer recovery. Maintaining 2 sources for peer recovery seems bad as discussed.

Reply:
As discussed - Leaving this as is for now rather in the plugin.